From 4601c66b7e3df055ba1f67beaf0dea7b4e817544 Mon Sep 17 00:00:00 2001
From: Arnaud Ferraris <arnaud.ferraris@gmail.com>
Date: Thu, 2 Jun 2022 23:05:59 +0200
Subject: [PATCH] cursor: avoid NULL pointer dereference

When touching the screen while blanked, phoc receives a `touch_up`
event, matching a previously discarded `touch_down` event (due to the
output being disabled).

This leads to a failure to retrieve the touch point while processing
`touch_up`, ultimately causing a segfault due to a NULL pointer
dereference.

Properly checking said pointer in order to avoid this crash.
---
 src/cursor.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/cursor.c b/src/cursor.c
index 49f2f098..cfa36b40 100644
--- a/src/cursor.c
+++ b/src/cursor.c
@@ -1030,6 +1030,11 @@ phoc_cursor_handle_touch_up (PhocCursor                *self,
   PhocTouchPoint *touch_point;
 
   touch_point = phoc_cursor_get_touch_point (self, event->touch_id);
+
+  /* Don't process unknown touch points */
+  if (!touch_point)
+    return;
+
   handle_gestures_for_event_at (self, touch_point->lx, touch_point->ly,
                                 PHOC_EVENT_TOUCH_END, event, sizeof (*event));
   phoc_cursor_remove_touch_point (self, event->touch_id);
-- 
GitLab

